src/porepy/grids/boundary_grid.py

```diff
@@ -56,24 +51,50 @@ def __init__(self, g: pp.Grid, name: Optional[str] = None) -> None:
         self._parent = g
         """Parent grid from which the boundary grid is constructed."""
 
-        self.cell_centers = g.face_centers[:, parent_boundary]
+        if g.dim == 0:
+            raise ValueError("Boundary grids are not supported for 0d grids.")
+        self.dim = g.dim - 1
+        """Dimension of the boundary grid."""
+
+        self.num_cells: int
+        """Number of cells in the boundary grid.
+
+        Will correspond to the number of faces on the domain boundary in the parent
+        grid. Initialized in :meth:`~compute_geometry`.
+
+        """
+        self._projections: sps.csr_matrix
+        """Projection matrix from the parent grid to the boundary grid.
+
+        Initialized in :meth:`~compute_geometry`.
+
+        """
+        self.cell_centers: np.ndarray
         """Cell centers of the boundary grid.
 
-        Subset of the face centers of the parent grid.
+        Subset of the face centers of the parent grid. Initialized in
+        :meth:`~compute_geometry`.
 
         """
 
-        if g.dim == 0:
-            raise ValueError("Boundary grids are not supported for 0d grids.")
-        self.dim = g.dim - 1
-        """Dimension of the boundary grid."""
+    def compute_geometry(self) -> None:
```

Comment:
As stated elsewhere, this is topology, not geometry. Consider renaming the method?

Reply:
Renaming it